spec/unit/shoulda/matchers/active_model/validate_acceptance_of_matcher_spec.rb

```diff
@@ -29,12 +29,67 @@
     end
   end
 
+  context 'when the writer method for the attribute changes incoming values' do
+    context 'and the matcher knows nothing of this' do
+      it 'raises a CouldNotSetAttributeError' do
+        model = define_model_validating_acceptance(
+          attribute_name: :terms_of_service
+        )
+
+        model.class_eval do
+          undef_method :terms_of_service=
+
+          def terms_of_service=(value)
+            if value
+              @terms_of_service = value
+            else
+              @terms_of_service = "something different"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.